hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorServiceProxy.java

```diff
@@ -0,0 +1,359 @@
+        Map<Member, IScheduledFuture<?>> futures = new HashMapAdapter<Member, IScheduledFuture<?>>();
+        for (Member member : members) {
+            TaskDefinition definition = new TaskDefinition(
+                    TaskDefinition.Type.WITH_REPETITION, name, adapter, initialDelay, 0, period, unit);
+
+            futures.put(member, submitOnMemberSync(name, new ScheduleTaskOperation(getName(), definition), member));
+        }
+
+        return futures;
+    }
+
+    @Override
+    public IScheduledFuture<?> getScheduledFuture(ScheduledTaskHandler handler) {
+        ScheduledFutureProxy proxy = new ScheduledFutureProxy(handler);
+        attachHazelcastInstance(proxy);
+        return proxy;
+    }
+
+    @Override
+    public <V> Map<Member, List<IScheduledFuture<V>>> getAllScheduledFutures() {
+        final long timeout = GET_ALL_SCHEDULED_TIMEOUT;
+
+        Map<Member, List<IScheduledFuture<V>>> tasks =
+                new LinkedHashMap<Member, List<IScheduledFuture<V>>>();
+
+        List<Member> members = new ArrayList<Member>(getNodeEngine().getClusterService().getMembers());
+        List<Future<List<ScheduledTaskHandler>>> futures = new ArrayList<Future<List<ScheduledTaskHandler>>>();
+        for (Member member : members) {
+            Address address = member.getAddress();
+
+            ICompletableFuture<List<ScheduledTaskHandler>> future =
+                    getOperationService().invokeOnTarget(getServiceName(),
+                            new GetAllScheduledOperation(getName()), address);
+
+            futures.add(future);
+        }
+
+        List<List<ScheduledTaskHandler>> resolvedFutures = new ArrayList<List<ScheduledTaskHandler>>(
+                returnWithDeadline(futures, timeout, TimeUnit.SECONDS));
+
+        for (int i = 0; i < resolvedFutures.size(); i++) {
+            Member member = members.get(i);
+            List<ScheduledTaskHandler> handlers = resolvedFutures.get(i);
+            List<IScheduledFuture<V>> scheduledFutures = new ArrayList<IScheduledFuture<V>>();
+
+            for (ScheduledTaskHandler handler : handlers) {
+                ScheduledFutureProxy proxy = new ScheduledFutureProxy(handler);
+                attachHazelcastInstance(proxy);
+                scheduledFutures.add(proxy);
+            }
+
+            if (!scheduledFutures.isEmpty()) {
+                tasks.put(member, scheduledFutures);
+            }
+        }
+
+        return tasks;
+    }
+
+    @Override
+    public void shutdown() {
+        NodeEngine nodeEngine = getNodeEngine();
+        Collection<Member> members = nodeEngine.getClusterService().getMembers();
+        OperationService operationService = nodeEngine.getOperationService();
+        Collection<Future> calls = new LinkedList<Future>();
+
+        for (Member member : members) {
+            if (member.localMember()) {
+                getService().shutdownExecutor(name);
```

Comment:
If you want to add the current member shutdown to the Future list, you can safely call the same `operationService.invokeOnTarget` method since it will internally check if you are invoking on the local member : com/hazelcast/spi/impl/operationservice/impl/Invocation.java:510